#!/usr/bin/env python
PACKAGE = "fav_control"

import roslib
roslib.load_manifest(PACKAGE)
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dynamic_reconfigure", bool_t, 0, "Enable Changing Parameters", False)
gen.add("controller_type",    int_t,    0, "0=SMC, 1=PD", 0,  0, 1)

pd = gen.add_group('pd')
pd.add("k_p", double_t, 0, "Parameter PD",    3.0, 0)
pd.add("k_d", double_t, 0, "Parameter  PD",    2.0, 0)

smc = gen.add_group('smc')
smc.add("alpha", double_t, 0, "Parameter SMC",    0.2, 0)
smc.add("Lambda", double_t, 0, "Parameter SMC",    1.5, 0)
smc.add("kappa", double_t, 0, "Parameter SMC",    2.5, 0)
smc.add("epsilon", double_t, 0, "Parameter SMC",    0.4, 0)

exit(gen.generate(PACKAGE, "control", "YawControl"))
